lms/djangoapps/instructor/views/serializers_v2.py

```diff
@@ -0,0 +1,337 @@
+                },
+                {
+                    'tab_id': 'grading',
+                    'title': _('Grading'),
+                    'url': f'{mfe_base_url}/instructor/{str(course_key)}/grading'
+                },
+                {
+                    'tab_id': 'cohorts',
+                    'title': _('Cohorts'),
+                    'url': f'{mfe_base_url}/instructor/{str(course_key)}/cohorts'
+                },
+            ])
+
+        if access['instructor'] and is_enabled_for_course(course_key):
+            tabs.append({
+                'tab_id': 'date_extensions',
+                'title': _('Date Extensions'),
+                'url': f'{mfe_base_url}/instructor/{str(course_key)}/date_extensions'
+            })
+
+        if access['data_researcher']:
+            tabs.append({
+                'tab_id': 'data_downloads',
+                'title': _('Data Downloads'),
+                'url': f'{mfe_base_url}/instructor/{str(course_key)}/data_downloads'
+            })
+
+        openassessment_blocks = modulestore().get_items(
+            course_key, qualifiers={'category': 'openassessment'}
+        )
+        # filter out orphaned openassessment blocks
+        openassessment_blocks = [
+            block for block in openassessment_blocks if block.parent is not None
+        ]
+        if len(openassessment_blocks) > 0 and access['staff']:
+            tabs.append({
+                'tab_id': 'open_responses',
+                'title': _('Open Responses'),
+                'url': f'{mfe_base_url}/instructor/{str(course_key)}/open_responses'
+            })
+
+        # Note: This is hidden for all CCXs
+        certs_enabled = CertificateGenerationConfiguration.current().enabled and not hasattr(course_key, 'ccx')
+        certs_instructor_enabled = settings.FEATURES.get('ENABLE_CERTIFICATES_INSTRUCTOR_MANAGE', False)
+
+        if certs_enabled and access['admin'] or (access['instructor'] and certs_instructor_enabled):
+            tabs.append({
+                'tab_id': 'certificates',
+                'title': _('Certificates'),
+                'url': f'{mfe_base_url}/instructor/{str(course_key)}/certificates'
+            })
+
+        user_has_access = any([
+            access['admin'],
+            CourseStaffRole(course_key).has_user(request.user),
+            access['instructor'],
+        ])
+        course_has_special_exams = course.enable_proctored_exams or course.enable_timed_exams
+        can_see_special_exams = course_has_special_exams and user_has_access and settings.FEATURES.get(
+            'ENABLE_SPECIAL_EXAMS', False)
+
+        if can_see_special_exams:
+            tabs.append({
+                'tab_id': 'special_exams',
+                'title': _('Special Exams'),
+                'url': f'{mfe_base_url}/instructor/{str(course_key)}/special_exams'
+            })
+
+        # Frontend requires tabs to be in a specific order
```

Comment:
```suggestion
        # We provide the tabs in a specific order based on how it was
        # historically presented in the frontend.  The frontend can use
        # this info or choose to ignore the ordering.
```

Remember that this API might outlive this implementation of the frontend and so more context here is useful to provide.

At first I thought that this ordering info should actually live in the frontend but then I realized that it would cause some coupling where if we ever change the names of these tabs in the backend, we'd have to synchronize the update to the frontend which doesn't make sense.  So I'm fine with leaving this here but wanted to suggest a better comment.

Reply:
I have updated comment and rebased this PR from master.